### PR TITLE
P-316 New Year - CryptoSummary

### PR DESCRIPTION
**Pull request by outofxxx**
[interface&example](https://www.notion.so/web3builders/NewYear-CryptoSummary-6da3808b0eab46309b2fc9acf3a99599)

### CODE REVIEW

File: tee-worker/litentry/core/credentials/src/nodereal/crypto_summary/mod.rs [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+						&CRYPTO_SUMMARY_NFT_ADDRESSES_ETH,
+						&nft_addresses,
+					)
+				}
+			}
+		}
+
+		// Update ETH and BNB
+		flag_bsc_token[11] = is_bnb_holder;
+		flag_eth_token[14] = is_eth_holder;
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/credentials/src/nodereal/crypto_summary/mod.rs:355]: I'm not a super fan of using such hardcoded index (same applies to the string tuple), but fine as long as it works for now

File: tee-worker/litentry/core/credentials/src/lib.rs [hunk 2]
```diff
@@ -320,10 +320,11 @@ impl Credential {
 			return Err(Error::EmptyCredentialIssuer)
 		}
 
-		let exported = vc.to_json()?;
-		if exported.len() > MAX_CREDENTIAL_SIZE {
-			return Err(Error::CredentialIsTooLong)
-		}
+		// TODO: Do we need to set size restrictions
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/credentials/src/lib.rs:323]: Did you find that the VC exceeds the limit this time?

File: tee-worker/litentry/core/credentials/src/nodereal/crypto_summary/mod.rs [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+impl SummaryHoldings {
+	pub fn is_empty(&self) -> bool {
+		self.summary.token.is_empty() && self.summary.nft.is_empty()
+	}
+
+	pub fn construct(bsc_token: &[bool], eth_token: &[bool], eth_nft: &[bool]) -> Self {
+		let bsc_token_list = Self::collect_list(&CRYPTO_SUMMARY_TOKEN_ADDRESSES_BSC, bsc_token);
+		let eth_token_list = Self::collect_list(&CRYPTO_SUMMARY_TOKEN_ADDRESSES_ETH, eth_token);
+		let eth_nft_list = Self::collect_list(&CRYPTO_SUMMARY_NFT_ADDRESSES_ETH, eth_nft);
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/credentials/src/nodereal/crypto_summary/mod.rs:184]: Shall we check if the list is empty?

**Review by Kailai-Wang — APPROVED** [review]
Thanks, the logic looks ok - the data needs to be tested for real inputs though

**Review by 0xverin — COMMENTED** [review]
Can you create an issue to add test cases in ts-tests for this vc? You can assign it to me. Thanks!

**Pull request merged**